src/config.rs

```diff
@@ -153,21 +225,67 @@ impl StarshipConfig {
             log::debug!("Using default config path: {}", config_path_str);
             config_path_str
         };
+        Some(home_file_path)
+    }
 
+    /// Create a config from a starship configuration file
+    fn config_from_file(file_path: &str) -> Option<Value> {
         let toml_content = match utils::read_file(&file_path) {
             Ok(content) => {
-                log::trace!("Config file content: \n{}", &content);
+                log::trace!("Config file {} content: \n{}", file_path, &content);
                 Some(content)
             }
             Err(e) => {
-                log::debug!("Unable to read config file content: \n{}", &e);
+                log::debug!("Unable to read config file {} content: \n{}", file_path, &e);
                 None
             }
         }?;
 
-        let config = toml::from_str(&toml_content).ok()?;
-        log::debug!("Config parsed: \n{:?}", &config);
-        Some(config)
+        match toml::from_str(&toml_content) {
+            Ok(config) => {
+                log::debug!("Config parsed: \n{:?}", &config);
+                Some(config)
+            }
+            Err(err) => {
+                log::warn!("Failed to parse config: \n{}", err);
+                None
+            }
+        }
+    }
+
+    /// Returns a deeply merged map where the values of map2 will overwrite the values in map1 if there is a collision
+    fn deep_merge(map1: Map<String, Value>, map2: Map<String, Value>) -> Map<String, Value> {
```

Comment:
Is a "deep merge" the expected behavior?
I can see it causing some unexpected behavior for users, rather than a regular merge.